generator/src/main.rs

```diff
@@ -1898,11 +1898,29 @@ impl Parser {
                 return None;
             }
             let ident = xr_var_name(&m.name);
+            // dbg!(self.structs.keys());
+            // dbg!(&m.ty);
             let (ty, value) = if m.ty == "XrBool32" {
                 (quote! { bool }, quote! { (self.0).#ident.into() })
             } else if self.handles.contains(&m.ty) {
                 let ty = xr_var_ty(m);
                 (quote! { sys::#ty }, quote! { (self.0).#ident })
+            } else if self.is_ptr_chain(&m.ty) {
+                let ty = xr_var_ty(m);
```

Comment:
@Ralith I'm having trouble replying to the original comment, but I wrapped the pointer chain reader in a Vec in response to your recommendation to expose an iterator.

This was simpler than generating code for a wrapper Iterator type. Let me know if this is what you had in mind!

Reply:
This produces a `Vec<*mut ViveTrackerPathsHTCX>`, which still requires the caller to write unsafe code to deref raw pointers. An ideal solution would return `impl Iterator<Item=ViveTrackerPathsHTCX>`, where `ViveTrackerPathsHTCX` is a handwritten type that omits the `s_type` and `pNext` members. It probably makes sense to write most, or even all, of the necessary code outside of the generator in a utility type/function, similar to the existing helpers in openxr/src/lib.rs.

Reply:
Don't worry too much about trying to implement a general solution if it becomes difficult; we don't know how often this pattern might come up in the future.

Reply:
I just realized that in [the spec](https://www.khronos.org/registry/OpenXR/specs/1.0/html/xrspec.html#XrViveTrackerPathsHTCX) `XrViveTrackerPathsHTCX.next` says
> next is NULL or a pointer to the next structure in a structure chain. No such structures are defined in core OpenXR or this extension.

I suppose this means the struct chain will only ever have one element. Does it make sense to just expose it as a single wrapper struct like below then?
```
pub struct ViveTrackerPaths {
    pub persistent: Path,
    pub role: Path,
}
```

Reply:
Oh, good, that's much simpler then. That would be ideal, yeah!

Reply:
added a manual wrapper in https://github.com/Ralith/openxrs/pull/97/commits/b287335bbdd253312d1db20faf119e445f80a997 , lmk what you think!